apiserver/facades/agent/uniter/uniter.go

```diff
@@ -453,15 +455,23 @@ func (u *UniterAPI) ClearResolved(ctx context.Context, args params.Entities) (pa
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
 			continue
 		}
-		err = apiservererrors.ErrPerm
-		if canAccess(tag) {
-			var unit *state.Unit
-			unit, err = u.getLegacyUnit(ctx, tag)
-			if err == nil {
-				err = unit.ClearResolved()
-			}
+		if !canAccess(tag) {
+			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
+			continue
+		}
+		unitName, err := coreunit.NewName(tag.Id())
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		err = u.resolveService.ClearResolved(ctx, unitName)
```

Comment:
I would still prefer to be resolving these names as UUIDs first, then passing them in, but we won't hold for that.